zerver/lib/actions.py

```diff
@@ -2226,20 +2226,20 @@ def do_add_reaction(
     emoji_code: str,
     reaction_type: str,
 ) -> None:
+    """Should be called while holding a SELECT FOR UPDATE lock
+    (e.g. via access_message(..., lock_message=True)) on the
+    Message row, to prevent race conditions.
+    """
```

Comment:
Thinking more about these comments, could we replace these with something like (pseudocode):

`assert message.row_is_locked`

so that the code is self-explanatory.  And we could potentially just set that property in `access_message` in the initial commit that adds support for `lock_message`.  (And have it be a default value that's false on the `Message` class, so one doesn't need to `getattr`?)

Reply:
Update: Mocking `send_event` is actually useful (https://github.com/zulip/zulip/pull/18559#issuecomment-854667223)